eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/Relayer.scala

```diff
@@ -148,51 +128,21 @@ class Relayer(nodeParams: NodeParams, router: ActorRef, register: ActorRef, paym
           PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, add.channelId, cmdFail)
       }
 
-    case Status.Failure(addFailed: AddHtlcFailed) => addFailed.origin match {
-      case Origin.Local(id, None) => log.error(s"received unexpected add failed with no sender (paymentId=$id)")
-      case Origin.Local(_, Some(sender)) => sender ! Status.Failure(addFailed)
-      case _: Origin.Relayed => channelRelayer forward Status.Failure(addFailed)
-      case Origin.TrampolineRelayed(htlcs, None) => log.error(s"received unexpected add failed with no sender (upstream=${htlcs.mkString(", ")}")
-      case Origin.TrampolineRelayed(_, Some(paymentSender)) => paymentSender ! Status.Failure(addFailed)
+    case r: RES_ADD_SETTLED[_, _] => r.origin match {
+      case _: Origin.Cold => postRestartCleaner ! r
+      case o: Origin.Hot => o.replyTo ! r
```

Comment:
Next logical step is to do that in the `Channel`, since it's completely stateless. Would that go to far?

Reply:
If we want to move to a full `request-reply` model and move that into the channel, we'll need to promote the `PostRestartHtlcCleaner`: it shouldn't be a child of the `Relayer` anymore but rather a top-level actor that the channel knows about. Maybe worth a try in a later PR?

Reply:
> we'll need to promote the PostRestartHtlcCleaner: it shouldn't be a child of the Relayer anymore

Indeed, see my early unfinished prototyped on branch `channel-postrestart`.

> Maybe worth a try in a later PR?

Okay